milabench/executors.py

```diff
@@ -0,0 +1,612 @@
+from __future__ import annotations
+
+import asyncio
+import os
+import json
+import socket
+from copy import deepcopy
+from hashlib import md5
+from typing import Dict, Generator, List, Tuple
+
+from . import pack
+from .alt_async import destroy
+from .fs import XPath
+from .merge import merge
+from .metadata import machine_metadata
+
+
+def clone_with(cfg, new_cfg):
+    return merge(deepcopy(cfg), new_cfg)
+
+
+async def force_terminate(pack, delay):
+    await asyncio.sleep(delay)
+    for proc in pack.processes:
+        ret = proc.poll()
+        if ret is None:
+            await pack.message(
+                f"Terminating process because it ran for longer than {delay} seconds."
+            )
+            destroy(proc)
+
+
+class Executor():
+    """Base class for an execution plan
+
+    Will reculsevly go through it's embedded `Executor` to build a command
```

Comment:
```suggestion
    Will recursively go through its embedded `Executor` to build a command
```

I assume. That was a particularly mangled up word ;)